hack/build-image/Dockerfile

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM --platform=linux/amd64 golang:1.21-bookworm
+FROM --platform=$BUILDPLATFORM golang:1.21-bookworm
```

Comment:
Why do we need to support multiple platforms?
And the change isn't enough as the binaries installed below are all linux amd64 version

Reply:
Apple Mx series desktop cannot build the build-image container without this change.